pkg/virt-launcher/virtwrap/converter/compute/clock.go

```diff
@@ -0,0 +1,122 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package compute
+
+import (
+	"strconv"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/virt-controller/watch/topology"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+type ClockDomainConfigurator struct{}
+
+func (c ClockDomainConfigurator) Configure(vmi *v1.VirtualMachineInstance, domain *api.Domain) error {
+	if vmi.Spec.Domain.Clock != nil {
+		clock := vmi.Spec.Domain.Clock
+		newClock := &api.Clock{}
+		err := convert_v1_Clock_To_api_Clock(clock, newClock)
+		if err != nil {
+			return err
+		}
+		domain.Spec.Clock = newClock
+	}
+
+	// Make use of the tsc frequency topology hint
+	if topology.IsManualTSCFrequencyRequired(vmi) {
+		freq := *vmi.Status.TopologyHints.TSCFrequency
+		clock := domain.Spec.Clock
+		if clock == nil {
+			clock = &api.Clock{}
+		}
+		clock.Timer = append(clock.Timer, api.Timer{Name: "tsc", Frequency: strconv.FormatInt(freq, 10)})
+		domain.Spec.Clock = clock
+	}
+
+	return nil
+}
+
+func convert_v1_Clock_To_api_Clock(source *v1.Clock, clock *api.Clock) error {
```

Comment:
According to golang standards, function names should be camelCase

```suggestion
func convertV1ClockToAPIClock(source *v1.Clock, clock *api.Clock) error {
```